src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -147,48 +300,182 @@ private void collectSymmetricalBinWidth(final JavaRDD<Integer> lengthRDD){
-        // TODO: fix second argument used in ctor, and potentially change how Histogram is constructed from TreeMap
-        htslibHistogram = new Histogram<>("insert_size", "All_Reads.fr_count");
-        for(final int size: hist.keySet()){
-            htslibHistogram.prefillBins(size);
-            htslibHistogram.increment(size, hist.get(size));
+        for(final Map.Entry<SamPairUtil.PairOrientation, List<Integer>> e : entry._2().entrySet()){
+            orientationToSortedMap.put(e.getKey(),
+                                       new TreeMap<>( e.getValue().stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting())) ));
         }
+        return new Tuple2<>(entry._1(), orientationToSortedMap);
     }
 
     /**
-     * Collect bulk information: pair orientation, sample name, library name and read groups.
+     * Write metrics and histograms to file.
+     * @param metricsFile File to write information to.
      */
-    private void collectLevelInfo(){
-        // TODO: change these, unchecked yet
-        metrics.PAIR_ORIENTATION = SamPairUtil.PairOrientation.FR;
-        metrics.SAMPLE = null;
-        metrics.LIBRARY = null;
-        metrics.READ_GROUP = null;
+    public void produceMetricsFile(final MetricsFile<InsertSizeMetrics, Integer> metricsFile) {
+
+        for(final SortedMap<GroupInfo, Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>>> maps: histogramsAndMetrics){
+            for(Map.Entry<GroupInfo, Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>>> groupEntry : maps.entrySet()){
+                final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> orientation2Tuple = groupEntry.getValue();
+                for(final Map.Entry<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> tuples : orientation2Tuple.entrySet()){
+                    metricsFile.addMetric(tuples.getValue()._2());
+                    metricsFile.addHistogram(tuples.getValue()._1());
+                }
+            }
+        }
     }
 
-    @VisibleForTesting
-    void produceMetricsFile(final MetricsFile<InsertSizeMetrics, Integer> metricsFile) {
-        metricsFile.addHistogram(htslibHistogram);
-        metricsFile.addMetric(metrics);
+    private static int convertMetricLevelToInt(final MetricAccumulationLevel level){
+        int i=0;
+        switch (level){
+            case ALL_READS:
+                break;
+            case SAMPLE:
+                i=1;
+                break;
+            case LIBRARY:
+                i=2;
+                break;
+            case READ_GROUP:
+                i=3;
+                break;
+        }
+        return i;
+    }
+
+    private static Comparator<GroupInfo> groupInfoComparator = new Comparator<GroupInfo>() {
+        @Override public int compare(final GroupInfo g1, final GroupInfo g2) {
+
+            final int first  = convertMetricLevelToInt(g1.level);
```

Comment:
Actually rather than converting to ints you should be able to just call compareTo on the levels themselves. Enums will compare themselves in the order in which they are declared in the Enum class so I think that will do what you want.